src/arviz_stats/regression.py

```diff
@@ -0,0 +1,107 @@
+
+    Parameters
+    ----------
+    data : DataTree or InferenceData
+        Input data. It should contain the posterior and the log_likelihood groups.
+    var_name : str
+        Name of the variable to compute the R² for.
+    data_pair : dict
+        Dictionary with the first element containing the posterior predictive name
+        and the second element containing the observed data variable name.
+    summary: bool
+        Whether to return a summary (default) or an array of R² samples.
+        The summary is a Pandas' series with a point estimate and a credible interval
+    point_estimate: str
+        The point estimate to compute. If None, the default value is used.
+        Defaults values are defined in rcParams["stats.point_estimate"]. Ignored if
+        summary is False.
+    ci_kind: str
+        The kind of credible interval to compute. If None, the default value is used.
+        Defaults values are defined in rcParams["stats.ci_kind"]. Ignored if
+        summary is False.
+    ci_prob: float
+        The probability for the credible interval. If None, the default value is used.
+        Defaults values are defined in rcParams["stats.ci_prob"]. Ignored if
+        summary is False.
+    round_to : int
+        Number of decimals used to round results. Defaults to 2. Use "none" to return raw numbers.
+
+    Returns
+    -------
+    Namedtuple or array
+
+    Examples
+    --------
+    Calculate R² samples for Bayesian regression models :
+
+    .. ipython::
+
+        In [1]: from arviz_stats import r2_score
+           ...: from arviz_base import load_arviz_data
+           ...: data = load_arviz_data('regression1d')
+           ...: r2_score(data)
+
+    References
+    ----------
+
+    .. [1] Gelman et al. *R-squared for Bayesian regression models*.
+        The American Statistician. 73(3) (2019). https://doi.org/10.1080/00031305.2018.1549100
+        preprint http://www.stat.columbia.edu/~gelman/research/published/bayes_R2_v3.pdf.
+    """
+    if point_estimate is None:
+        point_estimate = rcParams["stats.point_estimate"]
+    if ci_kind is None:
+        ci_kind = rcParams["stats.ci_kind"]
+    if ci_prob is None:
+        ci_prob = rcParams["stats.ci_prob"]
+
+    if data_pair is None:
+        obs_var_name = var_name
+        pred_var_name = var_name
+    else:
+        obs_var_name = list(data_pair.keys())[0]
+        pred_var_name = list(data_pair.values())[0]
+
+    y_true = extract(data, group="observed_data", var_names=obs_var_name, combined=False).values
+    y_pred = extract(data, group="posterior_predictive", var_names=pred_var_name).values.T
+
+    r_squared = array_stats.r2_score(y_true, y_pred)
+
+    if summary:
+        estimate = getattr(np, point_estimate)(r_squared).item()
+        c_i = getattr(array_stats, ci_kind)(r_squared, ci_prob)
+
+        r2_summary = namedtuple("R2", [point_estimate, f"{ci_kind}_lb", f"{ci_kind}_ub"])
```

Comment:
For `azs.summary`, we are using the notation f"{ci_kind}_", f"{ci_kind}_^". We can not do that for namedtuple as "^" is not supported. Whatever we use, it should be consistent.